test_runner/src/main/kotlin/ftl/run/platform/android/CreateAndroidTestContext.kt

```diff
@@ -99,11 +102,13 @@ private fun InstrumentationTestContext.calculateShards(
 ): InstrumentationTestContext = if (args.parameterizedTests.shouldShardSmartly()) {
     var flankTestMethods = getFlankTestMethods(testFilter, args.parameterizedTests)
     val parameterizedTests = flankTestMethods.filter { it.isParameterizedClass }
-    flankTestMethods = flankTestMethods.filterNot { it.isParameterizedClass }
+    val sdkSuppressTests = flankTestMethods.filter { it.sdkSuppressLevels != null }
+    flankTestMethods = flankTestMethods.filterNot { it.isParameterizedClass }.filterNot { it.sdkSuppressLevels != null }
     val shards = calculateParameterizedShards(flankTestMethods, args)
     val shardCount = if (args.parameterizedTests.isSingleParameterizedShard()) 1 else parameterizedTests.size
     val parameterizedShard = calculateParameterizedShards(parameterizedTests, args, shardCount)
-    shards.copy(shards = shards.shards + parameterizedShard.shards)
+    val sdkSuppressShard = calculateParameterizedShards(sdkSuppressTests, args, shardCount)
+    shards.copy(shards = shards.shards + parameterizedShard.shards + sdkSuppressShard.shards)
 } else calculateShardsNormally(args, testFilter)
```

Comment:
I have a feeling this logic should also be applied within the _calculateShardsNormally_ methodology.
For now it seems like it only works when we are sharding.

Reply:
I will investigate that case and let you know about the results

Reply:
@Sloox I have investigated this problem and looks that it is resolved by filtering tests during creating `FlankTestMethod`. I could not reproduce the reported problem, but I think that provided solution should fix the issue.

Reply:
Alright. Ill approve it then :)